### DISCUSSION TITLE
How to express LEFT JOIN

### DISCUSSION THREAD

**Discussion by tkawachi** [EvgSkv/logica#71] [Q&A] [answered]
I think a Conjunction is equivalent to an INNER JOIN in SQL, but how do I describe a LEFT JOIN?

**Comment by EvgSkv** [comment, marked as answer]
Left join is done via a combine expression. Like this:

```python

JoinedTable(key:, some_value:, joined_value:) :-
  MainTable(key:, some_value:),
  joined_value AnyValue= (x :- OuterTable(key:, value: x));
```

`AnyValue` should be used if it is known that `OuterTable` has no more than one row with the given key.
If it could have many values that `List` can be used to collect all of them, with appropriate further processing.

I have also added an [example](https://github.com/EvgSkv/logica/blob/main/examples/scripts/outer_join.l) in the scripts directory of assembling contacts with the outer join.

Please let me know if you have any questions or comments about this.

**Reply by tkawachi** [reply]
Thanks, I'll try it out. 😄
